cmd/config/config_check.go

```diff
@@ -0,0 +1,118 @@
+package config
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/calyptia/api/types"
+	cfg "github.com/calyptia/cli/config"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const KeyCliHealth = "cli_health"
+
+func NewCmdCheckInstall(c *cfg.Config) *cobra.Command {
+	return &cobra.Command{
+		Use:   "check-install",
+		Short: "Check the current configuration and report any issues",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			CheckInstall(c)
+			return nil
+		},
+	}
+}
+
+func CheckInstall(c *cfg.Config) {
+	ok := true
+	if !checkToken(c) {
+		ok = false
+	}
+	if !checkUrl(c) {
+		ok = false
+	}
+	if !checkSSL(c) {
+		ok = false
+	}
+	var k8sConfig *rest.Config
+	var pass bool
+	if k8sConfig, pass = checkK8sCredentials(); !pass {
+		ok = false
+	}
+	if !checkK8sHealth(k8sConfig) {
+		ok = false
+	}
+	if !ok {
+		fmt.Printf("Calyptia CLI is not ready to use\n")
+		return
+	}
+	err := c.LocalData.Save(KeyCliHealth, "ok")
+	if err != nil {
+		fmt.Printf("Calyptia CLI is ready to use but we couldnt save it health status\n")
+	}
+	fmt.Printf("Calyptia CLI is ready to use\n")
+
+}
+
+func checkToken(c *cfg.Config) bool {
+	if c.ProjectToken == "" {
+		fmt.Printf("You need to set a project token to use Calyptia CLI\n" +
+			"\tTip: you can use the flag --token or the command calyptia config set_token to set your project token\n")
+		return false
+	}
+	fmt.Printf("Token is set\n")
+	return true
+}
+
+func checkUrl(c *cfg.Config) bool {
+	ctx := context.Background()
+	_, err := c.Cloud.Environments(ctx, c.ProjectID, types.EnvironmentsParams{})
+	if err != nil {
```

Comment:
This simple error check is not good enough. If the token is incorrect, or cloud is down for some reason, you will get the "internet connection issues" anyway. Which is not fair.

We should check if `errors.Is(err, syscall.ECONNREFUSED)` to detect if the URL is wrong.
And hit the `/healthz` endpoint to check if the server is up.

Reply:
> Ensure the cloud API endpoint is reachable with the given token (this can be performed with any simple get/listing operation). ([issue#229](https://github.com/calyptia/cli/issues/229))
The idea is literally to test if we can do a get with the given token, but I improved the message now including more causes of the connection not working.

Reply:
@gabrielbussolo after this URL check just do quick connect to it with the token just to check validity of the token. You can do that in checkToken() fn.

I don't think that checkUrl is acutally going to help us at all because chances are that by doing `calyptia create core_instance k8s` core will be installed in a remote cluster.  What do you think?